libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentType.java

```diff
@@ -47,14 +49,26 @@ public String mediaType() {
         }
 
         @Override
-        public String subtype() {
+        public String formatPathParameter() {
             return "json";
         }
 
         @Override
         public XContent xContent() {
             return JsonXContent.jsonXContent;
         }
+
+        @Override
+        public Set<Tuple<String, Map<String, String>>> mediaTypeMappings() {
+            return Set.of(
+                Tuple.tuple("application/json", Map.of("charset", "UTF-8")),
```

Comment:
It's not clear from here if the "charset" is 'required', or if it is present only allows for these values. I think using a first class object (not a Tuple) would help with readability here.

Reply:
agree. It should look better with a class object here.